packages/sit-onyx/src/components/OnyxColorSchemeDialog/OnyxColorSchemeDialog.vue

```diff
@@ -0,0 +1,166 @@
+<script lang="ts" setup>
+import { computed, ref, watchEffect } from "vue";
+import { injectI18n } from "../../i18n";
+import OnyxButton from "../OnyxButton/OnyxButton.vue";
+import OnyxDialog from "../OnyxDialog/OnyxDialog.vue";
+import OnyxHeadline from "../OnyxHeadline/OnyxHeadline.vue";
+import OnyxRadioButton from "../OnyxRadioButton/OnyxRadioButton.vue";
+import autoImage from "./auto.svg?raw";
+import darkImage from "./dark.svg?raw";
+import lightImage from "./light.svg?raw";
+import type { ColorSchemeValue, OnyxColorSchemeDialogProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxColorSchemeDialogProps>(), {
+  open: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the color scheme should be changed.
+   */
+  "update:modelValue": [value: ColorSchemeValue];
+  /**
+   * Emitted when the dialog should be closed.
+   */
+  close: [];
+}>();
+
+const currentValue = ref(props.modelValue);
+watchEffect(() => (currentValue.value = props.modelValue));
+
+const { t } = injectI18n();
+
+const options = computed<
+  { value: ColorSchemeValue; description: string; image: string; label: string }[]
+>(() => {
+  return [
+    {
+      value: "auto",
+      image: autoImage,
+      label: t.value("colorScheme.auto.label"),
+      description: t.value("colorScheme.auto.description"),
+    },
+    {
+      value: "light",
+      image: lightImage,
+      label: t.value("colorScheme.light.label"),
+      description: t.value("colorScheme.light.description"),
+    },
+    {
+      value: "dark",
+      image: darkImage,
+      label: t.value("colorScheme.dark.label"),
+      description: t.value("colorScheme.dark.description"),
+    },
+  ];
+});
+
+const handleChange = (event: Event) => {
+  const target = event.target as HTMLInputElement;
+  currentValue.value = target.value as ColorSchemeValue;
+};
+
+const handleApply = () => {
+  emit("update:modelValue", currentValue.value);
+  emit("close");
+};
+</script>
+
+<template>
+  <OnyxDialog v-bind="props" :label="t('colorScheme.headline')" modal @close="emit('close')">
+    <form class="onyx-color-scheme-dialog" @submit.prevent="handleApply">
+      <div>
+        <OnyxHeadline is="h2"> {{ t("colorScheme.headline") }}</OnyxHeadline>
+        <span class="onyx-color-scheme-dialog__subtitle onyx-text">
+          {{ t("colorScheme.subtitle") }}
+        </span>
+      </div>
+
+      <fieldset class="onyx-color-scheme-dialog__list" @change="handleChange">
+        <div v-for="option in options" :key="option.value" class="onyx-color-scheme-dialog__option">
+          <!-- eslint-disable-next-line vue/no-v-html -->
+          <figure class="onyx-color-scheme-dialog__image" v-html="option.image"></figure>
+
+          <div>
```

Comment:
[optional ][cherry on top] if you want to flatten the dom a bit, instead of div-wrapping radio + p, you could use a grid layout on `onyx-color-scheme-dialog__option` and assign areas